### PR TITLE
User/sriramsriram msft/commits to upstream

### PR DESCRIPTION
**Pull request by sriramsriram-msft**

### CODE REVIEW

File: core/include/drivers/ls_dspi.h [hunk 1]
```diff
@@ -27,6 +27,13 @@
 #define DSPI_CTAR_A_DT(x)    (((x) & 0x0F) << 4)	/* DT [24-27] */
 #define DSPI_CTAR_BR(x)	     ((x) & 0x0F)	/* Baud Rate Scaler [28-31] */
 
+/* SPI mode flags */
```
Comment (jenswi-linaro) [inline core/include/drivers/ls_dspi.h:30]: Why are these needed here?

**Review by jenswi-linaro — COMMENTED** [review]
Comment for "core: drivers: ls_dspi: Add flushfifo and expose mode defines"

**Comment by jenswi-linaro** [comment]
For "core: drivers: Add missing argument in dt_map_dev function call"
please add a "Fixes:" tag.
By the way, this was obviously missed by CI. Can we tweak the CI
configuration (in a separate commit) to cover this too?

For "core: arch: arm: plat-bcm: Update broadcom platform file"
Please drop the "arch: " part for the prefix. In the commit description
you're covering the what, but not the why. Please explain why this is
needed.

For "core: arch: arm: plat-ls: Add CFG_WITH_ARM_TRUSTED_FW flag"
Please drop the "arch: " part for the prefix and add a "Fixes: " tag.

These commits are mostly unrelated to each other. That's not a problem, but
if one of the commits takes a bit longer to review all the other commits
will be delayed.

Please fix the checkpatch complaints.

File: core/include/drivers/ls_dspi.h (hunk 1 above)
Comment (sriramsriram-msft) [reply core/include/drivers/ls_dspi.h:30]: They are used in platform specific SPI configuration function.

**Comment by sriramsriram-msft** [comment]
Hi @jenswi-linaro, thanks for reviewing the pull request. Is there an example which explains how to tweak the CI configuration?

File: core/include/drivers/ls_dspi.h (hunk 1 above)
Comment (jenswi-linaro) [reply core/include/drivers/ls_dspi.h:30]: Not upstream, but not a big deal either. Please put this change in a separate commit explaining why it's needed.

File: core/drivers/crypto/caam/include/caam_desc_defines.h [hunk 2]
```diff
@@ -427,6 +427,9 @@
 #define AS_RNG_RESEED		0x2
 #define AS_RNG_UNINSTANTIATE	0x3
 
+/* predictive resistance */
+#define OP_ALG_PR_ON            0x2
```
Comment (clementfaure) [inline core/drivers/crypto/caam/include/caam_desc_defines.h:431]: This is already defined above
https://github.com/OP-TEE/optee_os/blob/6dc6fac61ed65b12a9196bd7f2d21814e73d79a4/core/drivers/crypto/caam/include/caam_desc_defines.h#L421-L423

File: core/include/drivers/ls_dspi.h (hunk 1 above)
Comment (sriramsriram-msft) [reply core/include/drivers/ls_dspi.h:30]: Done.

File: core/drivers/crypto/caam/include/caam_desc_defines.h (hunk 2 above)
Comment (sriramsriram-msft) [reply core/drivers/crypto/caam/include/caam_desc_defines.h]: Dropped the commit.

**Comment by sriramsriram-msft** [comment]
@jenswi-linaro 
By the way, this was obviously missed by CI. Can we tweak the CI
configuration (in a separate commit) to cover this too?

Our platform code is not upstream. Should I add it to lx2160-ardb platform?

**Comment by jenswi-linaro** [comment]
> @jenswi-linaro By the way, this was obviously missed by CI. Can we tweak the CI configuration (in a separate commit) to cover this too?
> 
> Our platform code is not upstream. Should I add it to lx2160-ardb platform?

That should be OK if that works.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
> core: drivers: Add missing argument in dt_map_dev function call

dt_map_dev()

> dt_map_dev function call is missing an argument which results in

dt_map_dev()

> compilation error. Add the missing argument.
>
> Fixes: 011c182 ("core: drivers: nxp: Add LX series SFP driver")
>

No blank line here please

> Signed-off-by: Sriram Sriram <sriramsriram@microsoft.com>
> 
> 
> core: arm: plat-bcm: Update broadcom platform file

`core: arm: plat-bcm: force CFG_CORE_ARM64_PA_BITS=48`

> 
> Set CFG_CORE_ARM64_PA_BITS in broadcom platform file. This fixes a crash

Broadcom

> when setting up memory addresses on the broadcom stingray NS3 platform

Broadcom

> 
> Signed-off-by: Stephen Carlson <stcarlso@microsoft.com>
> 
> 
> core: arm: plat-ls: Add CFG_WITH_ARM_TRUSTED_FW flag
> 
> If using ARM trusted FW, then the GIC initialization is done in BL31.

Arm Trusted Firmware-A

> Fixes: 2b9f239 ("plat-ls: Add support for armv8 platform flavours")

No blank line

> Signed-off-by: Carl Lamb <calamb@microsoft.com>
> 
> 
> core: drivers: ls_dspi: Add flush dspi module

DSPI

> Add flush dspi module

DSPI
It would be better to explain why, but no big deal.

> Signed-off-by: Carl Lamb <calamb@microsoft.com>
> 
> 
> core: drivers: Move SPI mode flag defines to header file

core: drivers: ls_dspi: Move SPI mode flag defines to header file

> SPI mode flags are used by NXP SoC specific functions for MS PTA
> 
> Signed-off-by: Sriram Sriram <sriramsriram@microsoft.com>

File: core/drivers/ls_dspi.c [hunk 3]
```diff
@@ -378,6 +378,18 @@ static void ls_dspi_end(struct spi_chip *chip)
...
+/*
+ * Flush DSPI module
+ * chip:	spi_chip instance
+ */
+static void ls_dspi_flush(struct spi_chip *chip)
+{
+	struct ls_dspi_data *data = container_of(chip, struct ls_dspi_data,
+						  chip);
+
+	dspi_flush_fifo(data);
```
Comment (sahilnxp) [inline core/drivers/ls_dspi.c:383]: Make `dspi_flush_fifo `as static also as it gets exposed via `ls_dspi_ops`.
Also remove it from `ls_dspi.h`

File: core/arch/arm/plat-ls/main.c [hunk 4]
```diff
@@ -215,7 +215,9 @@ void main_init_gic(void)
 	gic_init(gic_base + gicc_offset, gic_base + gicd_offset);
```
Comment (sahilnxp) [inline core/arch/arm/plat-ls/main.c:215]: This `gic_init() `should also be in same flag `#if !defined(CFG_WITH_ARM_TRUSTED_FW)` 
BTW, Did you faced any issue without this patch ?

**Review by sahilnxp — CHANGES_REQUESTED** [review]

**Comment by sriramsriram-msft** [comment]
> > @jenswi-linaro By the way, this was obviously missed by CI. Can we tweak the CI configuration (in a separate commit) to cover this too?
> > Our platform code is not upstream. Should I add it to lx2160-ardb platform?
> 
> That should be OK if that works.

Will create a separate PR for this.

**Comment by sriramsriram-msft** [comment]
> > core: drivers: Add missing argument in dt_map_dev function call
> 
> dt_map_dev()
> 
> > dt_map_dev function call is missing an argument which results in
> 
> dt_map_dev()
> 
> > compilation error. Add the missing argument.
> > Fixes: [011c182](https://github.com/OP-TEE/optee_os/commit/011c182a606b95697862181bc24e628cab39fcc6) ("core: drivers: nxp: Add LX series SFP driver")
> 
> No blank line here please
> 
> > Signed-off-by: Sriram Sriram [sriramsriram@microsoft.com](mailto:sriramsriram@microsoft.com)
> > core: arm: plat-bcm: Update broadcom platform file
> 
> `core: arm: plat-bcm: force CFG_CORE_ARM64_PA_BITS=48`
> 
> > Set CFG_CORE_ARM64_PA_BITS in broadcom platform file. This fixes a crash
> 
> Broadcom
> 
> > when setting up memory addresses on the broadcom stingray NS3 platform
> 
> Broadcom
> 
> > Signed-off-by: Stephen Carlson [stcarlso@microsoft.com](mailto:stcarlso@microsoft.com)
> > core: arm: plat-ls: Add CFG_WITH_ARM_TRUSTED_FW flag
> > If using ARM trusted FW, then the GIC initialization is done in BL31.
> 
> Arm Trusted Firmware-A
> 
> > Fixes: [2b9f239](https://github.com/OP-TEE/optee_os/commit/2b9f239231751444a6b5eda3d933ecb24e679fdf) ("plat-ls: Add support for armv8 platform flavours")
> 
> No blank line
> 
> > Signed-off-by: Carl Lamb [calamb@microsoft.com](mailto:calamb@microsoft.com)
> > core: drivers: ls_dspi: Add flush dspi module
> 
> DSPI
> 
> > Add flush dspi module
> 
> DSPI It would be better to explain why, but no big deal.
> 
> > Signed-off-by: Carl Lamb [calamb@microsoft.com](mailto:calamb@microsoft.com)
> > core: drivers: Move SPI mode flag defines to header file
> 
> core: drivers: ls_dspi: Move SPI mode flag defines to header file
> 
> > SPI mode flags are used by NXP SoC specific functions for MS PTA
> > Signed-off-by: Sriram Sriram [sriramsriram@microsoft.com](mailto:sriramsriram@microsoft.com)

Fixed the issues.

File: core/arch/arm/plat-ls/main.c (hunk 4 above)
Comment (sriramsriram-msft) [reply core/arch/arm/plat-ls/main.c:215]: https://github.com/OP-TEE/optee_os/commit/0ee3f52ed191d21fb12ed5832d4b7929ac7001e4 removed it. We didn't see any issues without the patch because we don't use GIC before it is reinitialized by optee.

**Comment by sahilnxp** [comment]
Just a minor comment on commit title `core: drivers: ls_sfp: Add missing argument in dt_map_dev() function call` in place of `core: drivers: Add missing argument in dt_map_dev() function call`

With this taken care, `Acked-by: Sahil Malhotra <sahil.malhotra@nxp.com>` for all below commits.

`core: drivers: ls_dspi: Make dspi_flush_fifo() static`
`core: drivers: ls_dspi: Move SPI mode flag defines to header file`
`core: drivers: ls_dspi: Add flush DSPI module`
`core: arm: plat-ls: Add CFG_WITH_ARM_TRUSTED_FW flag`
`core: drivers: Add missing argument in dt_map_dev() function call`

**Comment by sriramsriram-msft** [comment]
> Just a minor comment on commit title `core: drivers: ls_sfp: Add missing argument in dt_map_dev() function call` in place of `core: drivers: Add missing argument in dt_map_dev() function call`
> 
> With this taken care, `Acked-by: Sahil Malhotra <sahil.malhotra@nxp.com>` for all below commits.
> 
> `core: drivers: ls_dspi: Make dspi_flush_fifo() static` `core: drivers: ls_dspi: Move SPI mode flag defines to header file` `core: drivers: ls_dspi: Add flush DSPI module` `core: arm: plat-ls: Add CFG_WITH_ARM_TRUSTED_FW flag` `core: drivers: Add missing argument in dt_map_dev() function call`

Fixed it.
